### PR TITLE
feat: generate wallet from xpub

### PR DESCRIPTION
**Pull request by r4mmer**

### CODE REVIEW

File: src/new/wallet.js [hunk 1]
```diff
@@ -854,7 +857,7 @@ class HathorWallet extends EventEmitter {
     const options = Object.assign({ pinCode: null, password: null }, optionsParams);
     const pinCode = options.pinCode || this.pinCode;
     const password = options.password || this.password;
-    if (!pinCode) {
+    if (!this.xpub && !pinCode) {
```
Comment (pedroferreira1) [inline src/new/wallet.js:860]: There are more methods in this wallet class where you must have the xpriv to run, e.g. send tx, create token, mint, ... all those methods should be protected and throw an error if you try to call them and you have only the xpub.

**Review by pedroferreira1 — CHANGES_REQUESTED** [review]

File: src/wallet.js [hunk 2]
```diff
@@ -630,13 +651,17 @@ const wallet = {
    * @inner
    */
   changePin(oldPin, newPin) {
+    const accessData = this.getWalletAccessData();
+
+    if (accessData['from_xpub']) {
```
Comment (msbrogli) [inline src/wallet.js:656]: Create a method `isFromXpub()` and use it in all these verifications.

File: src/wallet.js [hunk 3]
```diff
@@ -1511,6 +1536,9 @@ const wallet = {
    */
   getWalletWords(password) {
     const accessData = this.getWalletAccessData();
+    if (accessData['from_xpub']) {
```
Comment (msbrogli) [inline src/wallet.js:1539]: Use `isFromXpub()` here.

File: src/wallet.js [hunk 4]
```diff
@@ -2435,6 +2463,11 @@ const wallet = {
    */
   getXprivKey(pin) {
     const accessData = this.getWalletAccessData();
+
+    if (accessData['from_xpub']) {
```
Comment (msbrogli) [inline src/wallet.js:2467]: Use `isFromXpub()` here.

**Review by msbrogli — APPROVED** [review]

File: src/new/wallet.js (hunk 1 above)
Comment (r4mmer) [reply src/new/wallet.js:879]: Done

File: src/wallet.js (hunk 2 above)
Comment (r4mmer) [reply src/wallet.js]: Done

**Review by msbrogli — APPROVED** [review]

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**